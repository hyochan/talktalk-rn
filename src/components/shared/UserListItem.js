// @flow
import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TouchableOpacity,
} from 'react-native';
import Icon5, { FA5Style } from 'react-native-vector-icons/FontAwesome5';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { ratio, colors } from '../../utils/Styles';
import { User as Friend } from '../../models/User';

type Styles = {
  container: ViewStyle,
  wrapper: ViewStyle,
  img: ImageStyle,
  txt: TextStyle,
  txtRight: TextStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  wrapper: {
    backgroundColor: 'white',
    height: 80,
    borderBottomWidth: 1,
    borderColor: 'rgb(245,245,245)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingHorizontal: 20,
  },
  img: {
    width: 40,
    height: 40,
  },
  txt: {
    marginLeft: 12,
    width: 100,
    fontSize: 14,
    color: colors.dusk,
  },
  txtRight: {
    position: 'absolute',
    right: 20,
    fontSize: 12,
    color: colors.dusk,
    maxWidth: 134.2,
    borderWidth: 0.3,
    borderColor: colors.paleGray,
    paddingHorizontal: 8,
    paddingVertical: 4,
  },
});

type Props = {
  style: ViewStyle;
  user: Friend;
  onPress?: () => void;
  onLongPress?: () => void;
};

type State = {};

function Shared(props: Props, state: State) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        activeOpacity={0.5}
        onPress={props.onPress}
        onLongPress={props.onLongPress}
      >
        <View style={props.style}>
          {
            props.user.photoURL
              ? <Image style={styles.img} source={props.user.photoURL}/>
              : <Icon5 name="meh" size={40} color={colors.dusk} light/>
          }
          <Text style={styles.txt}>{props.user.displayName}</Text>
          {
            props.user.statusMsg
              ? <Text style={styles.txtRight}>{props.user.statusMsg}</Text>
              : <View/>
          }
        </View>
      </TouchableOpacity>
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  user: {
    uid: '',
    displayName: '',
    photoURL: null,
    statusMsg: '',
    isOnline: '',
    friends: '',
    Chatrooms: '',
    created: '',
    updated: '',
  },
};
export default Shared;
