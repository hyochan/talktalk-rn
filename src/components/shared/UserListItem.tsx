// @flow
import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  TouchableOpacity,
  ViewStyle,
  ImageStyle,
  TextStyle,
} from 'react-native';
import Icon5, { FA5Style } from 'react-native-vector-icons/FontAwesome5';

import { ratio, colors } from '../../utils/Styles';
import { User as Friend } from '../../models/User';

interface IStyles {
  container: ViewStyle;
  wrapper: ViewStyle;
  img: ImageStyle;
  txt: TextStyle;
  txtRight: TextStyle;
}

const styles: IStyles = StyleSheet.create({
  container: {
    width: '100%',
  },
  wrapper: {
    backgroundColor: 'white',
    height: 80,
    borderBottomWidth: 1,
    borderColor: 'rgb(245,245,245)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingHorizontal: 20,
  },
  img: {
    width: 40,
    height: 40,
  },
  txt: {
    marginLeft: 12,
    width: 100,
    fontSize: 14,
    color: colors.dusk,
  },
  txtRight: {
    position: 'absolute',
    right: 20,
    fontSize: 12,
    color: colors.dusk,
    maxWidth: 134.2,
    borderWidth: 0.3,
    borderColor: colors.paleGray,
    paddingHorizontal: 8,
    paddingVertical: 4,
  },
});

interface IProps {
  testID?: string;
  testObj?: any;
  style: ViewStyle;
  user: Friend;
  onPress?: () => void;
  onLongPress?: () => void;
}

interface IState {}

function Shared(props: IProps, state: IState) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        testID='press_id'
        activeOpacity={0.5}
        onPress={props.onPress}
        onLongPress={props.onLongPress}
      >
        <View style={props.style}>
          {
            props.user.photoURL
              ? <Image style={styles.img} source={{ uri: props.user.photoURL }}/>
              : <Icon5 name='meh' size={40} color={colors.dusk} light/>
          }
          <Text style={styles.txt}>{props.user.displayName}</Text>
          {
            props.user.statusMsg
              ? <Text style={styles.txtRight}>{props.user.statusMsg}</Text>
              : <View/>
          }
        </View>
      </TouchableOpacity>
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  user: {
    uid: '',
    displayName: '',
    photoURL: null,
    statusMsg: '',
    isOnline: '',
    friends: '',
    Chatrooms: '',
    created: '',
    updated: '',
  },
};
export default Shared;
