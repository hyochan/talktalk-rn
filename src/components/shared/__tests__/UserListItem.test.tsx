import 'react-native';
import * as React from 'react';
import UserListItem from '../UserListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

let cnt = 0;
const onPress = () => {
  cnt++;
};
const component: React.ReactElement = <UserListItem onPress={onPress}/>;

describe('[UserListItem] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[UserListItem] interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let testingLib: any;

  beforeAll(() => {
    rendered = renderer.create(component);
    testingLib = render(component);
  });

  it('should fireEvent when peer image is pressed', () => {
    const touchBtn = testingLib.getByTestId('press_id');
    fireEvent(touchBtn, 'press');
    expect(cnt).toEqual(1);
  });
});
