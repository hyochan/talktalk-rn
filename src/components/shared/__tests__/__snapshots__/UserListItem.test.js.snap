// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UserListItem] rendering test renders as expected 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="press_id"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "rgb(245,245,245)",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "flex-start",
          "paddingHorizontal": 20,
        }
      }
    >
      <Unknown
        color="rgb(65,77,107)"
        light={true}
        name="meh"
        size={40}
      />
      <Text
        style={
          Object {
            "color": "rgb(65,77,107)",
            "fontSize": 14,
            "marginLeft": 12,
            "width": 100,
          }
        }
      >
        
      </Text>
      <View />
    </View>
  </View>
</View>
`;
